File: Python/problem3.py
import math


def is_prime(number: int) -> bool:
    for i in range(2, math.ceil(math.sqrt(number)), 1):
        if number % i == 0:
            return False
    return True


def solve(number: int) -> int:
    factor: int = 0
    for i in range(1, math.ceil(math.sqrt(number)), 1):
        if number % i == 0:
            if is_prime(i):
                factor = i
    return factor


if __name__ == "__main__":
    print(solve(600851475143))

File: C++/Problem3.c
#include <iostream>
#include <math.h>
using namespace std;

bool isPrime(int number)
{
  for (int i = 3; i <=  round(sqrt(number))+1; i += 2)
    if (number % i == 0) return false;
  return true;
}

int main()
{
  long number = 600851475143;
  int max_factor = 0;

  for (int i = 3; i <= round(sqrt(number)) +  1; i += 2)
  {
    if (number % i == 0)
    {
      if (isPrime(i))
      {
        max_factor = i;
      }
    }
  }

  cout << max_factor << endl;
}
